### PR TITLE
Fix python performance tests on Jenkins

### PR DESCRIPTION
**Pull request by DariuszAniszewski**
* use an absolute path to read Beam version for Python SDK
* build python SDK on Jenkins and enable the test

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by DariuszAniszewski** [comment]
Run Seed Job

**Comment by DariuszAniszewski** [comment]
Run Python Performance Test

**Comment by DariuszAniszewski** [comment]
Run Seed Job

**Comment by DariuszAniszewski** [comment]
Run Python Performance Test

**Comment by DariuszAniszewski** [comment]
Run Seed Job

**Comment by DariuszAniszewski** [comment]
Run Seed Job

**Comment by DariuszAniszewski** [comment]
Run Python Performance Test

**Comment by DariuszAniszewski** [comment]
Run Seed Job

**Comment by DariuszAniszewski** [comment]
Run Python Performance Test

**Comment by markflyhigh** [comment]
Thanks to fix this test. Looks like `test_wordcount_fnapi_it` is invoked in Performance Test which should use `test_wordcount_it` as default instead. The problem comes from [this line](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py#L60) which includes all test methods in that module.

Before fix it on Perfkit, I think you can set `--beam_it_class=apache_beam.examples.wordcount_it_test:WordCountIT.test_wordcount_it` ([defined here](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/linux_benchmarks/beam_integration_benchmark.py#L62)) to command line to invoke `test_wordcount_it` only.

**Comment by markflyhigh** [comment]
https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/1742 can fix the problem on Perfkit side.

**Comment by DariuszAniszewski** [comment]
@markflyhigh Thanks for hints - I tried to follow your idea but even with running only `test_wordcount_it` test on Jenkins is still timeouting.

This problem seems to have some more complicated nature or I'm missing something obvious here.
The exact same command (exact to the point of GCS buckets and project) ran from my computer finishes within several minutes - both `test_wordcount_fnapi_it` and `test_wordcount_it`. What's more, [the last job that succedded](https://builds.apache.org/job/beam_PerformanceTests_Python/749/console) also runs both of them.

I've bisecting repo and found commit that started to putting Dataflow jobs into some infinite loop. The commit that "broke" the test is [this one](https://github.com/apache/beam/commit/3dc643fbc02536d24938a6c86046b387d835607d).
It was hard for me to belive that this commit was a problem, but I started to running the pipelines and they in fact stuck. I was partially reverting pieces of the commit and from my tests it seems that adding this check was the culptit:
```
    if num_retries > 10000:
      raise ValueError('num_retries parameter cannot exceed 10000.')
``` 

I have no answers why this is a problem. I can only imagine that this exception is catched somehow and check is retried, and it fails, so it's retried, and we have infinite loop of retrying.

The other open question is why test timeouts on Jenkins and finishes outside of it. Maybe there is some infra issue there?

**Comment by markflyhigh** [comment]
Your fix lgtm. I tried your changes on my local machine and run this benchmark with same flags (except GCP project and GCS directory), and it also finishes. So it's pretty wired that timeout on Jenkins.

Wondering if the pipeline is actually submitted to Dataflow or not.

**Comment by markflyhigh** [comment]
Run Seed Job

**Comment by markflyhigh** [comment]
Run Python Performance Test

File: .test-infra/jenkins/CommonJobProperties.groovy [hunk 1]
```diff
@@ -322,6 +322,9 @@ class CommonJobProperties {
         // Install job requirements for Python SDK.
```
Comment (markflyhigh) [inline .test-infra/jenkins/CommonJobProperties.groovy]: can you also add `pip install --upgrade tox==3.0.0 grpcio-tools==1.3.5` for Python SDK? This shows in Python gradle `setupVirtualenv` ([here](https://github.com/apache/beam/blob/master/sdks/python/build.gradle#L38)) which is used to run all Pre/Post-commit.

**Comment by DariuszAniszewski** [comment]
Closing this in favor of #6142

**Pull request closed without merging**